sdk/monitor/monitor-query/src/metricsClient.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { TokenCredential } from "@azure/core-auth";
+import {
+  bearerTokenAuthenticationPolicy,
+  createPipelineFromOptions,
+  PipelineOptions
+} from "@azure/core-http";
+
+import {
+  GetMetricDefinitionsOptions,
+  GetMetricDefinitionsResponse,
+  GetMetricNamespacesOptions,
+  GetMetricNamespacesResponse,
+  QueryMetricsOptions,
+  QueryMetricsResponse
+} from "./models/publicMetricsModels";
+
+import {
+  KnownApiVersion20170501Preview as MetricsApiVersion,
+  MonitorManagementClient as GeneratedMetricsClient
+} from "./generated/metrics/src";
+import {
+  KnownApiVersion20170501Preview as MetricDefinitionsApiVersion,
+  MetricsDefinitionsClient as GeneratedMetricsDefinitionsClient
+} from "./generated/metricsdefinitions/src";
+import {
+  KnownApiVersion20171201Preview as MetricNamespacesApiVersion,
+  MetricsNamespacesClient as GeneratedMetricsNamespacesClient
+} from "./generated/metricsnamespaces/src";
+import {
+  convertRequestForMetrics,
+  convertRequestOptionsForMetricsDefinitions,
+  convertResponseForMetricNamespaces,
+  convertResponseForMetrics,
+  convertResponseForMetricsDefinitions
+} from "./internal/modelConverters";
+
+export interface MetricsClientOptions extends PipelineOptions {
+  /** server parameter */
+  $host?: string;
+  /** Overrides client endpoint. */
+  endpoint?: string;
+}
+
+export class MetricsClient {
+  private _metricsClient: GeneratedMetricsClient;
+  private _definitionsClient: GeneratedMetricsDefinitionsClient;
+  private _namespacesClient: GeneratedMetricsNamespacesClient;
+
+  constructor(tokenCredential: TokenCredential, options?: MetricsClientOptions) {
+    const bearerTokenPolicy = bearerTokenAuthenticationPolicy(
+      tokenCredential,
+      formatScope(options?.endpoint)
+    );
+
+    const serviceClientOptions = {
+      ...createPipelineFromOptions(options || {}, bearerTokenPolicy),
+      $host: options?.$host,
+      endpoint: options?.endpoint
+    };
+
+    this._metricsClient = new GeneratedMetricsClient(
+      tokenCredential,
+      MetricsApiVersion.TwoThousandSeventeen0501Preview,
```

Comment:
We really need to fix this enum name because it is 🍌 🍌 🍌

Reply:
It's totally generated code. I'm a bit insulted you think I'd create an enum like that! (checks code).

Reply:
Oh I know it's generated, my remark was about how we should fix the codegen, not that you did anything wrong. :)